model_analyzer/config/input/config_command.py

```diff
@@ -94,41 +95,164 @@ def set_config_values(self, args):
             this exception
         """
 
-        # Config file has been specified
+        yaml_config = self._load_yaml_config(args)
+        self._check_for_illegal_config_settings(args, yaml_config)
+        self._set_field_values(args, yaml_config)
+        self._preprocess_and_verify_arguments()
+        self._autofill_values()
+
+    def _load_yaml_config(self, args: Namespace) -> Optional[Dict[str, List]]:
         if 'config_file' in args:
             yaml_config = self._load_config_file(args.config_file)
             YamlConfigValidator.validate(yaml_config)
         else:
             yaml_config = None
 
+        return yaml_config
+
+    def _check_for_illegal_config_settings(
```

Comment:
What happens if the user does `run_config_search_disable: True` in the yaml or in the CLI? Do we ignore brute/quick and just run the default config?

Reply:
We ignore the search disable. What is the correct thing to do here? Make this an error?

Reply:
I've made this an error. Please review the new code.

Reply:
Seems like we should honor the disable, but that would require larger changes (maybe in the factory?). Can you file a story to follow up?

Reply:
I've filed TMA-939